File: API/_archive/SOAP/golang/01_inbuilt_basic.go
// BASIC EXAMPLE: Go Inbuilt net/http + encoding/xml
// Demonstrates raw SOAP Envelope construction via structs
package main

import (
	"bytes"
	"encoding/xml"
	"fmt"
	"io"
	"net/http"
)

type Envelope struct {
	XMLName xml.Name `xml:"soap:Envelope"`
	Soap    string   `xml:"xmlns:soap,attr"`
	Body    Body
}

type Body struct {
	XMLName       xml.Name       `xml:"soap:Body"`
	NumberToWords *NumberToWords `xml:"NumberToWords"`
}

type NumberToWords struct {
	Xmlns  string `xml:"xmlns,attr"`
	UbiNum int    `xml:"ubiNum"`
}

func main() {
	env := Envelope{
		Soap: "http://schemas.xmlsoap.org/soap/envelope/",
		Body: Body{
			NumberToWords: &NumberToWords{
				Xmlns:  "http://www.dataaccess.com/webservicesserver/",
				UbiNum: 500,
			},
		},
	}
	
	payload, _ := xml.Marshal(env)

	resp, err := http.Post(
		"http://www.dataaccess.com/webservicesserver/numberconversion.wso",
		"text/xml; charset=utf-8",
		bytes.NewReader(payload),
	)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	body, _ := io.ReadAll(resp.Body)
	fmt.Printf("Raw XML Response:\n%s\n", string(body))
}

File: API/_archive/SOAP/python/01_inbuilt_basic.py
# BASIC EXAMPLE: Python Inbuilt urllib + xml.etree
# Demonstrates raw XML Envelope construction and HTTP POST
import urllib.request
import xml.etree.ElementTree as ET

url = "http://www.dataaccess.com/webservicesserver/numberconversion.wso"

# Construct the raw XML SOAP Envelope
soap_body = """<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <NumberToWords xmlns="http://www.dataaccess.com/webservicesserver/">
      <ubiNum>500</ubiNum>
    </NumberToWords>
  </soap:Body>
</soap:Envelope>"""

req = urllib.request.Request(url, data=soap_body.encode("utf-8"), headers={
    "Content-Type": "text/xml; charset=utf-8"
})

response = urllib.request.urlopen(req)
root = ET.fromstring(response.read())

# Parse XML namespaces
ns = {'m': 'http://www.dataaccess.com/webservicesserver/'}
result = root.find('.//m:NumberToWordsResult', ns).text

print(f"Result from SOAP service: {result}")
